pydicom/dataset.py

```diff
@@ -1854,15 +1857,19 @@ def overlay_array(self, group: int) -> "numpy.ndarray":
                 )
                 last_exception = exc
 
-        logger.info(
-            "Unable to decode the overlay data using the following handlers: "
-            "{}. Please see the list of supported Transfer Syntaxes in the "
-            "pydicom documentation for alternative packages that might "
-            "be able to decode the data"
-            .format(", ".join([str(hh) for hh in available_handlers]))
-        )
-
-        raise last_exception  # type: ignore[misc]
+        if last_exception is None:
+            pass  # FIXME: raise some exception?
```

Comment:
Same here. It does not help readability to introduce branches for conditions which cannot happen.

Reply:
It can happen if `available_handlers` is empty.

Reply:
But if available_handlers cannot be empty, then I agree. I'll look into it.

Reply:
Perhaps I'll open (yet another) bug report against LGTM.